File: onyx-database-tests/test/memory/system/SystemEntityTest.java
package memory.system;

import category.InMemoryDatabaseTests;
import com.onyx.entity.SystemEntity;
import com.onyx.exception.OnyxException;
import com.onyx.exception.InitializationException;
import com.onyx.persistence.query.Query;
import com.onyx.persistence.query.QueryCriteria;
import com.onyx.persistence.query.QueryCriteriaOperator;
import junit.framework.Assert;
import org.junit.After;
import org.junit.Before;
import org.junit.Test;
import org.junit.experimental.categories.Category;

import java.io.IOException;
import java.util.List;

/**
 * Created by timothy.osborn on 3/7/15.
 */
@Category({ InMemoryDatabaseTests.class })
public class SystemEntityTest extends memory.base.BaseTest
{

    @Before
    public void before() throws InitializationException, InterruptedException
    {
        initialize();
    }

    /**
     * This should delete the database after the test category has finished
     *
     * @throws OnyxException
     * @throws IOException
     */
    @After
    public void after() throws IOException
    {
        shutdown();
        deleteDatabase();
    }

    @Test
    public void testQuerySystemEntities() throws OnyxException
    {
        Query query = new Query();
        query.setEntityType(SystemEntity.class);
        query.setCriteria(new QueryCriteria("name", QueryCriteriaOperator.NOT_EQUAL, QueryCriteria.NULL_STRING_VALUE));

        List<SystemEntity> results = manager.executeQuery(query);
        Assert.assertTrue(results.size() > 0);
    }
}

File: onyx-database-tests/test/embedded/system/SystemEntityTest.kt
package embedded.system

import category.EmbeddedDatabaseTests
import com.onyx.entity.SystemEntity
import com.onyx.exception.OnyxException
import com.onyx.exception.InitializationException
import com.onyx.persistence.query.Query
import com.onyx.persistence.query.QueryCriteria
import com.onyx.persistence.query.QueryCriteriaOperator
import embedded.base.BaseTest
import junit.framework.Assert
import org.junit.After
import org.junit.Before
import org.junit.Test
import org.junit.experimental.categories.Category

import java.io.IOException

/**
 * Created by timothy.osborn on 3/7/15.
 */
@Category(EmbeddedDatabaseTests::class)
class SystemEntityTest : BaseTest() {

    @Before
    @Throws(InitializationException::class, InterruptedException::class)
    fun before() {
        initialize()
    }

    /**
     * This should delete the database after the test category has finished
     *
     * @throws OnyxException
     * @throws IOException
     */
    @After
    @Throws(IOException::class)
    fun after() {
        shutdown()
        BaseTest.deleteDatabase()
    }

    @Test
    @Throws(OnyxException::class)
    fun testQuerySystemEntities() {
        val query = Query()
        query.entityType = SystemEntity::class.java
        query.criteria = QueryCriteria("name", QueryCriteriaOperator.NOT_NULL)

        val results = manager.executeQuery<SystemEntity>(query)
        Assert.assertTrue(results.size > 0)
    }
}
